ompi/mca/io/romio341/configure.m4

```diff
@@ -99,14 +99,20 @@ AC_DEFUN([MCA_ompi_io_romio341_CONFIG],[
                           . 3rd-party/romio341/localdefs
                           io_romio341_LIBS="$LIBS"
                           LIBS="$io_romio341_save_LIBS"
-                          OPAL_3RDPARTY_SUBDIRS="$OPAL_3RDPARTY_SUBDIRS romio341"
                           OPAL_3RDPARTY_DIST_SUBDIRS="$OPAL_3RDPARTY_DIST_SUBDIRS romio341"
 
-                          echo "ROMIO distribution configured successfully"
-                          $1],
+                          echo "ROMIO distribution configured successfully"],
                          [AS_IF([test "$enable_io_romio" = "yes"],
                                 [AC_MSG_ERROR([ROMIO distribution did not configure successfully])],
-                                [AC_MSG_WARN([ROMIO distribution did not configure successfully])])
-                          $2])])])
+                                [AC_MSG_WARN([ROMIO distribution did not configure successfully])])])
+
+                   AS_IF([test "$io_romio341_happy" = "1"],
+                         [AC_CHECK_HEADERS([stdatomic.h],
+                                           [OPAL_3RDPARTY_SUBDIRS="$OPAL_3RDPARTY_SUBDIRS romio341"],
+                                           [io_romio341_happy=0])])
+
+                   AS_IF([test "io_romio341_happy" = "1"],
+                         [$1],
+                         [$2])])])
```

Comment:
Shouldn't the "ROMIO distribution configured successfully / did not configure successfully" messages be moved down here?  I ask because it is now possible for you to see "ROMIO distribution configured successfully", but then stdatomic.h is not found, so the ROMIO component actually *fails* to configure successfully.

Reply:
The thing is unless we patch ROMIO, `stdatomic.h` will not be tested there, so we do it here.
So yes, 3rd party ROMIO will `configure` successfully, but `io/romio341` is the one trapping the issue and will discard itself so we do not build 3rd party ROMIO if we know it will fail.

The error message can be counter intuitive, but on the upside, I think we can run `make dist` even on RHEL7.

Reply:
Ok, I see what you're saying: you want a message indicating that ROMIO itself -- as opposed to the ROMIO 3.1.4 OMPI component -- configured successfully.  Fair enough.  How about a compromise: add a AC_MSG_WARN message in the stdatomic.h fail case indicating that even though the ROMIO distribution itself configured successfully, this failed test will disqualify the ROMIO 3.1.4 OMPI component blah blah blah...